### PR TITLE
Various maintenance

### LINKED ISSUE #34: Set `MPLBACKEND=Agg` in headless Linux
This is currently handled via xtgeo, but it should (and will need to be done) here, particularly when `xtgeo.plot` is deprecated.

### LINKED ISSUE #38: Directly accessing `.dataframe` causes xtgeo to issue warnings against itself
A few places directly access the internal dataframe which we have set a future deprecation warning for. This means using a quickplot function from xtgeo will issue an xtgeo deprecation warning. These should use the new getters and setters instead.

### PR DESCRIPTION
**Pull request by mferrera**
Resolves #34 
Resolves #38 
Resolves GitHub actions using Node 16 https://github.com/equinor/atlas/issues/36

### CODE REVIEW

File: src/xtgeoviz/plot/xsection.py [hunk 1]
```diff
@@ -379,13 +379,13 @@ def plot_well(
...
+        wo.set_dataframe(dfr[dfr["Z_TVDSS"] > self._zmin])
 
         # Create a relative XYLENGTH vector (0.0 where well starts)
         wo.create_relative_hlen()
 
-        dfr = wo.dataframe
+        dfr = wo.get_dataframe()
         if dfr.empty:
             self._showok = False
             return
```
Comment (jcrivenaes) [inline src/xtgeoviz/plot/xsection.py:391]: Here we could simplify even more:

`if wo.get_dataframe(copy=False).empty:  ...`

File: tests/test_plot/test_map.py [hunk 2]
```diff
@@ -35,9 +35,8 @@ def test_map_plot_with_points(tmpdir, generate_plot):
 
     mypoints = xtgeo.points_from_surface(mysurf)
 
-    df = mypoints.dataframe.copy()
-    df = df[::20]
-    mypoints.dataframe = df
+    df = mypoints.get_dataframe()
+    mypoints.set_dataframe(df[::29])
```
Comment (jcrivenaes) [inline tests/test_plot/test_map.py:39]: Any reasons for going from `::20` to `::29`?

**Review by jcrivenaes — APPROVED** [review]
👍 . One could consider use more of `points.get_dataframe(copy=False)` in order to return the current underlying dataframe object instead of a copy (when this feels safe). Getting a copy will be slower and more memory demanding.

File: tests/test_plot/test_map.py (hunk 2 above)
Comment (mferrera) [reply tests/test_plot/test_map.py:39]: Nice catch, not sure how this happened.

File: src/xtgeoviz/plot/xsection.py (hunk 1 above)
Comment (mferrera) [reply src/xtgeoviz/plot/xsection.py:391]: It seems we actually need to refresh the local variable again after running `create_relative_hlen()`

**Pull request merged**